velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -737,44 +737,6 @@ PlanBuilder& PlanBuilder::assignUniqueId(
 }
 
 namespace {
-class HashPartitionFunctionSpec : public core::PartitionFunctionSpec {
```

Comment:
Since we deleted these, should we also delete PlanBuilder::registerSerDe method?

```
// static
void PlanBuilder::registerSerDe() {
  auto& registry = DeserializationWithContextRegistryForSharedPtr();

  registry.Register(
      "HashPartitionFunctionSpec", HashPartitionFunctionSpec::deserialize);
  registry.Register(
      "RoundRobinPartitionFunctionSpec",
      RoundRobinPartitionFunctionSpec::deserialize);
}
```

Reply:
No actually, we remove those just to get rid of duplication, we only want to have a centralized place to store those partition function spec
But still we need to `registerSerDe` (but it will register our new version of `HashPartitionFunctionSpec` and `RoundRobinPartitionFunctionSpec`) 

Otherwise, tests like `PlanNodeToStringTest` etc will fail because it couldn't find the Serialize and Deserialize function

Reply:
PlanBuilder is a test utility. It should not be responsible for registering "production" classes. We need to find a different place to register these.

Reply:
@kewang1024 can we put all the partition function implementations (except the gather function which is defined in core/PlanNode) into one file partitionFunction.h/cc and provide a global utility
```
void registerPartitionFunctionSerDe() {
  auto& registry = DeserializationWithContextRegistryForSharedPtr();

  registry.Register(
      "HashPartitionFunctionSpec", HashPartitionFunctionSpec::deserialize);
  registry.Register(
      "RoundRobinPartitionFunctionSpec",
      RoundRobinPartitionFunctionSpec::deserialize); 
   registry.Register(
      "GatherPartitionFunctionSpec", GatherPartitionFunctionSpec::deserialize); <== we can move the gather function register here as well
   hive function register.
}

And replace PlanBuilder::registerSerDe() with registerPartitionFunctionSerDe in current unit tests.
```
@mbasmanova  @kewang1024 WDYT? Thanks!

Reply:
Do you mean all related classes like `HashPartitionFunction`, `RoundRobinPartitionFunction` etc?

Reply:
@xiaoxmeng This should work. HiveXxxSpec can be registered by HiveConnectorFactory.